src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -210,6 +210,13 @@ public sealed partial class TokenCache : ITokenCacheInternal
                     ITokenCacheInternal tokenCacheInternal = this;
                     if (tokenCacheInternal.IsTokenCacheSerialized())
                     {
+                        DateTimeOffset? cacheExpiry = null;
+
+                        if (requestParams.IsClientCredentialRequest)
```

Comment:
Let's try to make the code more robust here. This is hardcoding the fact that client_creds flow is currentlty the only one where RTs are not involved. 

I think the code would be more robust if the conditition was "there are no RTs in this cache". Hence you can call `_accessor.GetAllRefreshTokens().Any()`